### PR TITLE
Dart 1.0 fixes

### PR DESCRIPTION
**Pull request by sdeleuze**
I only did basic testing so I think there is some bugs left.

I had to remove IFrame support (was not used in previous code) since last Dart SDK does't give access to IFrame ReadyState and only authorize postMessage communication.

Thanks for your feedback ....

### CODE REVIEW

File: example/echo.dart [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+InputElement inp  = querySelector('#first input');
+FormElement form = querySelector('#first form');
 
 print(m, [p = '']) {
   if(!p.isEmpty) {
-    p = JSON.stringify(p);
+    p = JSON.encode(p);
   }
-  div.elements
+  div.children
```
Comment (nelsonsilva) [inline example/echo.dart]: div
..append(...)
..append(...)

File: lib/src/events.dart [hunk 2]
```diff
@@ -11,9 +11,11 @@ class Events {
 
   Map<String, ListenerList> _listeners;
 
-  Events() : _listeners = <ListenerList>{};
+  Events() : _listeners = <String, ListenerList>{};
```
Comment (nelsonsilva) [inline lib/src/events.dart:14]: Will switch to Streams after merging this PR for consistency

File: lib/src/ajax.dart [hunk 3]
```diff
@@ -18,6 +18,7 @@ class AbstractXHRObject implements event.Emitter<XHREvents> {
   XHREvents on = new XHREvents();
 
   HttpRequest xhr;
+  StreamSubscription orschss;
```
Comment (nelsonsilva) [inline lib/src/ajax.dart]: these variable names seem confusing. Let's just call them "changeSubscription" or something like that

File: lib/src/ajax.dart [hunk 4]
```diff
@@ -109,7 +110,7 @@ class AbstractXHRObject implements event.Emitter<XHREvents> {
 
 class XHRCorsObject extends AbstractXHRObject {
    XHRCorsObject(method, url, payload, {noCredentials, headers} )  {
-    utils.delay(()=>_start(method, url, payload, noCredentials: false));
+    Timer.run(() =>_start(method, url, payload, noCredentials: false));
```
Comment (nelsonsilva) [inline lib/src/ajax.dart:113]: Why did you remove this from utils ?

File: lib/src/client.dart [hunk 5]
```diff
@@ -136,7 +136,7 @@ class Client implements event.Emitter<SockJSEvents> {
   }
 
   _dispatchHeartbeat() {
-    if (readyState !== OPEN) {
+    if (!identical(readyState, OPEN)) {
```
Comment (nelsonsilva) [inline lib/src/client.dart]: no identical here just an extra "=" inherited form the JS code just remove it: readyState != OPEN
There are very few use cases for identical so you probably just don't need it, ever.

File: lib/src/transport/sender.dart [hunk 6]
```diff
@@ -62,82 +62,6 @@ class BufferedSender {
   }
 }
 
-class JsonPGenericSender {
```
Comment (nelsonsilva) [inline lib/src/transport/sender.dart:65]: I know this will be available in the vcs history but can we keep it here with a TODO at the beginning ?

File: lib/src/transport/websocket.dart [hunk 7]
```diff
@@ -2,9 +2,11 @@ part of sockjs_client;
 
 class WebSocketTransport {
   Client ri;
-  var url;
+  String url;
 
   WebSocket ws;
+  StreamSubscription omhss;
```
Comment (nelsonsilva) [inline lib/src/transport/websocket.dart]: again... these are just difficult to remember.. camel case something short like "messageSubscription" and "closeSubscription"

File: lib/src/utils.dart [hunk 8]
```diff
@@ -44,11 +43,11 @@ amendUrl(String url) {
     }
 
     //  '//abc' --> 'http://abc'
-    if (url.indexOf('//') === 0) {
+    if (identical(url.indexOf('//'), 0)) {
```
Comment (nelsonsilva) [inline lib/src/utils.dart]: == instead of identical

File: lib/src/utils.dart [hunk 9]
```diff
@@ -44,11 +43,11 @@ amendUrl(String url) {
...
         url = "${dl.protocol}$url";
     }
     // '/abc' --> 'http://localhost:80/abc'
-    if (url.indexOf('/') === 0) {
+    if (identical(url.indexOf('/'), 0)) {
```
Comment (nelsonsilva) [inline lib/src/utils.dart]: == instead of identical

File: pubspec.yaml [hunk 10]
```diff
@@ -1,4 +1,8 @@
 name:  sockjs_client
-version: 0.1.0
-author: Nelson Silva <nelson.silva@inevo.pt>
+version: 0.2.0
+authors:
+- Nelson Silva <nelson.silva@inevo.pt>
+- Sébastien Deleuze
```
Comment (nelsonsilva) [inline pubspec.yaml:5]: Do you have an email to add here ?

**Comment by sdeleuze** [comment]
I am going to update this PR according to your feedbacks, thanks

File: pubspec.yaml [hunk 11]
```diff
@@ -1,4 +1,8 @@
...
 description:  A SockJS client in Dart
+dependencies:
```
Comment (nelsonsilva) [inline pubspec.yaml:7]: Let's make this safer now that Dart 1.0 is out:

environment:
  sdk: '>=1.0.0 <2.0.0'
dependencies:
  browser: '>=0.9.0 <1.0.0'

**Comment by nelsonsilva** [comment]
@sdeleuze Thx. I just added some quick remarks.
Also, you should try to avoid working on the master branch. Just set it to track "upstream" (or whatever you call the original repo) and work on your own custom branch. This way it's easier to rebase if needed and you avoid merge problems. I'll hold off any changes to my master to prevent conflicts until this gets merged.

File: lib/src/ajax.dart (hunk 4 above)
Comment (sdeleuze) [inline lib/src/ajax.dart:113]: I removed utils.delay because I thought it was useful in Javascript but found that Dart Timer.run and constructor were simple enough to use them directly, without addind a not very useful wrapper.

Is it ok for you or do you want me to restore utils.delay ?

Comment (nelsonsilva) [inline lib/src/ajax.dart:113]: makes sense. I'm ok with it. The initial idea was to make it as close as possible to the original version to make it easier to update to newer versions but i guess it won't happen any time soon so it's not a problem...

File: pubspec.yaml (hunk 10 above)
Comment (sdeleuze) [inline pubspec.yaml:5]: This is to avoid spam. Dart documentation specify that author could be provided without email (http://pub.dartlang.org/doc/pubspec.html#authorauthors) I hope this is ok for you.

Comment (nelsonsilva) [inline pubspec.yaml:5]: sure. no prob. thought you forgot it.

**Comment by sdeleuze** [comment]
Thanks for your feedbacks.
I have updated my PR and will use my own branch for next ones.
Let me know if this is ok for you ...

**Comment by nelsonsilva** [comment]
Thx @sdeleuze. I'll take the opportunity to go through the code and try to fix a couple more things.

**Comment by sdeleuze** [comment]
You're welcome !

After your fixes I will continue testing (with Spring 4.0 SockJS backend mostly) and contributions.
Feel free to create issues with the tasks you have identified, I am really interested to make this lib production ready.

My contributions will be mostly on the client side for the moment.
